src/commons/workspace/WorkspaceReducer.ts

```diff
@@ -841,6 +843,18 @@ const newWorkspaceReducer = createReducer(defaultWorkspaceManager, builder => {
           breakpointSteps: action.payload.breakpointSteps
         }
       };
+    })
+    .addCase(toggleUsingUpload, (state, action) => {
+      const { workspaceLocation } = action.payload;
+      if (workspaceLocation === 'playground' || workspaceLocation === 'sicp') {
+        state[workspaceLocation].usingUpload = action.payload.usingUpload;
+      }
+    })
+    .addCase(uploadFiles, (state, action) => {
+      const { workspaceLocation } = action.payload;
```

Comment:
`workspaceLocation` should not be accessed via destructuring of `action.payload`. Refer to the other functions for how (i.e. `getWorkspaceLocation`).